lab-davide/lib/storage.js

```diff
@@ -0,0 +1,73 @@
+
+'use strict';
+
+const debug = require('debug')('note:storage');
+const AppError = require('./app-error');
+exports.pool = {};
+
+exports.setItem = function(schema, item){
+  debug('setItem');
+  return new Promise((resolve, reject) => {
+    if (!item.id) {
+      var err = AppError.error400('storage setItem requires id');
+      return reject(err);
+    }
+    if (!this.pool[schema]) this.pool[schema] = {};
+    this.pool[schema][item.id] = item;
+    resolve(item);
+  });
+};
+
+exports.fetchItem = function(schema, id){
+  debug('fetchItem');
+  return new Promise((resolve, reject) => {
+    if (!this.pool[schema]){
+      var err = AppError.error404('storage schema not found');
```

Comment:
This is a little goofy--declaring a variable inside of an if block, and then occasionally using it inside of another if block. I would either do:
```javascript
let err = null;
if (condition) {
  err = AppError.error404();
}
if (other condition) {
  err = AppError.error400();
}
```
or something like this: 
```javascript
if (condition) {
  let err = AppError.error404();
}
if (other condition) {
  let err = AppError.error400();
}
```